<Type Name="SemaphoreFullException" FullName="System.Threading.SemaphoreFullException">
  <Metadata>
    <Meta Name="ms.openlocfilehash" Value="3376f73eb9294bdba4a519f82e567653975fccf6" />
    <Meta Name="ms.sourcegitcommit" Value="5a49536d99d2d0b54e4cb7280870903e043272df" />
    <Meta Name="ms.translationtype" Value="HT" />
    <Meta Name="ms.contentlocale" Value="it-IT" />
    <Meta Name="ms.lasthandoff" Value="07/03/2018" />
    <Meta Name="ms.locfileid" Value="37548679" />
  </Metadata>
  <TypeSignature Language="C#" Value="public class SemaphoreFullException : SystemException" />
  <TypeSignature Language="ILAsm" Value=".class public auto ansi serializable beforefieldinit SemaphoreFullException extends System.SystemException" />
  <TypeSignature Language="DocId" Value="T:System.Threading.SemaphoreFullException" />
  <TypeSignature Language="VB.NET" Value="Public Class SemaphoreFullException&#xA;Inherits SystemException" />
  <TypeSignature Language="C++ CLI" Value="public ref class SemaphoreFullException : SystemException" />
  <TypeSignature Language="F#" Value="type SemaphoreFullException = class&#xA;    inherit SystemException" />
  <AssemblyInfo>
    <AssemblyName>System.Threading</AssemblyName>
    <AssemblyVersion>4.0.0.0</AssemblyVersion>
    <AssemblyVersion>4.0.10.0</AssemblyVersion>
    <AssemblyVersion>4.1.0.0</AssemblyVersion>
    <AssemblyVersion>4.1.1.0</AssemblyVersion>
  </AssemblyInfo>
  <AssemblyInfo>
    <AssemblyName>mscorlib</AssemblyName>
    <AssemblyVersion>2.0.5.0</AssemblyVersion>
    <AssemblyVersion>4.0.0.0</AssemblyVersion>
  </AssemblyInfo>
  <AssemblyInfo>
    <AssemblyName>netstandard</AssemblyName>
    <AssemblyVersion>2.0.0.0</AssemblyVersion>
  </AssemblyInfo>
  <AssemblyInfo>
    <AssemblyName>System</AssemblyName>
    <AssemblyVersion>2.0.0.0</AssemblyVersion>
  </AssemblyInfo>
  <Base>
    <BaseTypeName>System.Exception</BaseTypeName>
    <BaseTypeName FrameworkAlternate="netframework-4.5">System.SystemException</BaseTypeName>
  </Base>
  <Interfaces />
  <Attributes>
    <Attribute>
      <AttributeName>System.Runtime.CompilerServices.TypeForwardedFrom("System, Version=2.0.0.0, Culture=Neutral, PublicKeyToken=b77a5c561934e089")</AttributeName>
    </Attribute>
    <Attribute>
      <AttributeName>System.Runtime.InteropServices.ComVisible(false)</AttributeName>
    </Attribute>
  </Attributes>
  <Docs>
    <summary>
      <span data-ttu-id="71613-101">Eccezione generata quando il metodo <see cref="Overload:System.Threading.Semaphore.Release" /> viene chiamato su un semaforo il cui conteggio ha già raggiunto il valore massimo.</span>
      <span class="sxs-lookup">
        <span data-stu-id="71613-101">The exception that is thrown when the <see cref="Overload:System.Threading.Semaphore.Release" /> method is called on a semaphore whose count is already at the maximum.</span>
      </span>
    </summary>
    <remarks>
      <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="71613-102">Il conteggio sul semaforo viene decrementato ogni volta che un thread accede al semaforo e incrementato quando un thread rilascia il semaforo.</span><span class="sxs-lookup"><span data-stu-id="71613-102">The count on a semaphore is decremented each time a thread enters the semaphore, and incremented when a thread releases the semaphore.</span></span> <span data-ttu-id="71613-103">Quando il conteggio è zero, le richieste successive blocca fino a quando gli altri thread rilasciano il semaforo.</span><span class="sxs-lookup"><span data-stu-id="71613-103">When the count is zero, subsequent requests block until other threads release the semaphore.</span></span> <span data-ttu-id="71613-104">Quando tutti i thread hanno rilasciato il semaforo, il conteggio viene raggiunto il valore massimo specificato quando è stato creato il semaforo.</span><span class="sxs-lookup"><span data-stu-id="71613-104">When all threads have released the semaphore, the count is at the maximum value specified when the semaphore was created.</span></span> <span data-ttu-id="71613-105">Se un errore di programmazione, un thread chiama il <xref:System.Threading.Semaphore.Release%2A?displayProperty=nameWithType> metodo a questo punto, un <xref:System.Threading.SemaphoreFullException> viene generata un'eccezione.</span><span class="sxs-lookup"><span data-stu-id="71613-105">If a programming error causes a thread to call the <xref:System.Threading.Semaphore.Release%2A?displayProperty=nameWithType> method at this point, a <xref:System.Threading.SemaphoreFullException> is thrown.</span></span>  
  
> [!NOTE]
>  <span data-ttu-id="71613-106">Il <xref:System.Threading.Semaphore> classe non impone l'identità thread alle chiamate per il <xref:System.Threading.WaitHandle.WaitOne%2A?displayProperty=nameWithType> e <xref:System.Threading.Semaphore.Release%2A?displayProperty=nameWithType> metodi.</span><span class="sxs-lookup"><span data-stu-id="71613-106">The <xref:System.Threading.Semaphore> class does not enforce thread identity on calls to the <xref:System.Threading.WaitHandle.WaitOne%2A?displayProperty=nameWithType> and <xref:System.Threading.Semaphore.Release%2A?displayProperty=nameWithType> methods.</span></span> <span data-ttu-id="71613-107">Non è necessario per lo stesso thread che ha chiamato <xref:System.Threading.WaitHandle.WaitOne%2A> chiamare <xref:System.Threading.Semaphore.Release%2A>.</span><span class="sxs-lookup"><span data-stu-id="71613-107">It is not necessary for the same thread that called <xref:System.Threading.WaitHandle.WaitOne%2A> to call <xref:System.Threading.Semaphore.Release%2A>.</span></span>  
  
 <span data-ttu-id="71613-108"><xref:System.Threading.SemaphoreFullException> non indica necessariamente un problema con il codice in cui si è verificata l'eccezione.</span><span class="sxs-lookup"><span data-stu-id="71613-108"><xref:System.Threading.SemaphoreFullException> does not necessarily indicate a problem with the code where the exception occurred.</span></span> <span data-ttu-id="71613-109">Si consideri lo scenario seguente: Thread A e B accedano a un semaforo che presenta un numero massimo di due.</span><span class="sxs-lookup"><span data-stu-id="71613-109">Consider the following scenario: Thread A and thread B enter a semaphore that has a maximum count of two.</span></span> <span data-ttu-id="71613-110">Un errore di programmazione nel thread B fa sì che venga chiamato <xref:System.Threading.Semaphore.Release%2A> due volte, in modo che il conteggio sul semaforo è completo.</span><span class="sxs-lookup"><span data-stu-id="71613-110">A programming error in thread B causes it to call <xref:System.Threading.Semaphore.Release%2A> twice, so that the count on the semaphore is full.</span></span> <span data-ttu-id="71613-111">Di conseguenza, quando il thread chiama infine <xref:System.Threading.Semaphore.Release%2A>, un <xref:System.Threading.SemaphoreFullException> viene generata un'eccezione.</span><span class="sxs-lookup"><span data-stu-id="71613-111">As a result, when thread A eventually calls <xref:System.Threading.Semaphore.Release%2A>, a <xref:System.Threading.SemaphoreFullException> is thrown.</span></span>  
  
 <span data-ttu-id="71613-112">Per un elenco dei valori iniziali delle proprietà di un'istanza della classe <xref:System.Threading.SemaphoreFullException>, vedere il costruttore <xref:System.Threading.SemaphoreFullException.%23ctor>.</span><span class="sxs-lookup"><span data-stu-id="71613-112">For a list of initial property values for an instance of the <xref:System.Threading.SemaphoreFullException> class, see the <xref:System.Threading.SemaphoreFullException.%23ctor> constructor.</span></span>  
  
   
  
## Examples  
 <span data-ttu-id="71613-113">Esempio di codice seguente viene illustrato come un errore di programmazione in un thread può causare un <xref:System.Threading.SemaphoreFullException> in un altro thread: due thread immettere un semaforo.</span><span class="sxs-lookup"><span data-stu-id="71613-113">The following code example shows how a programming error in one thread can lead to a <xref:System.Threading.SemaphoreFullException> in another thread: Two threads enter a semaphore.</span></span> <span data-ttu-id="71613-114">Il secondo thread rilascia il semaforo due volte, mentre il primo thread è ancora in esecuzione l'attività.</span><span class="sxs-lookup"><span data-stu-id="71613-114">The second thread releases the semaphore twice, while the first thread is still executing its task.</span></span> <span data-ttu-id="71613-115">Quando il primo thread termina e rilascia il semaforo, il conteggio del semaforo è completo e viene generata un'eccezione.</span><span class="sxs-lookup"><span data-stu-id="71613-115">When the first thread finishes and releases the semaphore, the semaphore count is already full and an exception is thrown.</span></span>  
  
 [!code-cpp[System.Threading.SemaphoreFullException#1](~/samples/snippets/cpp/VS_Snippets_CLR_System/system.Threading.SemaphoreFullException/CPP/source.cpp#1)]
 [!code-csharp[System.Threading.SemaphoreFullException#1](~/samples/snippets/csharp/VS_Snippets_CLR_System/system.Threading.SemaphoreFullException/CS/source.cs#1)]
 [!code-vb[System.Threading.SemaphoreFullException#1](~/samples/snippets/visualbasic/VS_Snippets_CLR_System/system.Threading.SemaphoreFullException/VB/source.vb#1)]  
  
 ]]></format>
    </remarks>
    <altmember cref="T:System.Threading.Semaphore" />
  </Docs>
  <Members>
    <MemberGroup MemberName=".ctor">
      <AssemblyInfo>
        <AssemblyName>System.Threading</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.10.0</AssemblyVersion>
        <AssemblyVersion>4.1.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Docs>
        <summary>
          <span data-ttu-id="71613-116">Inizializza una nuova istanza della classe <see cref="T:System.Threading.SemaphoreFullException" />.</span>
          <span class="sxs-lookup">
            <span data-stu-id="71613-116">Initializes a new instance of the <see cref="T:System.Threading.SemaphoreFullException" /> class.</span>
          </span>
        </summary>
      </Docs>
    </MemberGroup>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public SemaphoreFullException ();" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor() cil managed" />
      <MemberSignature Language="DocId" Value="M:System.Threading.SemaphoreFullException.#ctor" />
      <MemberSignature Language="VB.NET" Value="Public Sub New ()" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; SemaphoreFullException();" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Threading</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.10.0</AssemblyVersion>
        <AssemblyVersion>4.1.0.0</AssemblyVersion>
        <AssemblyVersion>4.1.1.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>System</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters />
      <Docs>
        <summary>
          <span data-ttu-id="71613-117">Inizializza una nuova istanza della classe <see cref="T:System.Threading.SemaphoreFullException" /> con i valori predefiniti.</span>
          <span class="sxs-lookup">
            <span data-stu-id="71613-117">Initializes a new instance of the <see cref="T:System.Threading.SemaphoreFullException" /> class with default values.</span>
          </span>
        </summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="71613-118">Questo costruttore inizializza il <xref:System.Exception.Message%2A> proprietà della nuova istanza di un messaggio fornito dal sistema che descrive l'errore, ad esempio "Somma il conteggio specificato per il semaforo può causare al superamento del limite massimo."</span><span class="sxs-lookup"><span data-stu-id="71613-118">This constructor initializes the <xref:System.Exception.Message%2A> property of the new instance to a system-supplied message that describes the error, such as "Adding the given count to the semaphore would cause it to exceed its maximum count."</span></span> <span data-ttu-id="71613-119">Questo messaggio prende in considerazione le impostazioni cultura correnti del sistema.</span><span class="sxs-lookup"><span data-stu-id="71613-119">This message takes into account the current system culture.</span></span>  
  
 <span data-ttu-id="71613-120">Nella tabella riportata di seguito sono illustrati i valori iniziali delle proprietà per un'istanza della classe <xref:System.Threading.SemaphoreFullException>.</span><span class="sxs-lookup"><span data-stu-id="71613-120">The following table shows the initial property values for an instance of the <xref:System.Threading.SemaphoreFullException> class.</span></span>  
  
|<span data-ttu-id="71613-121">Proprietà</span><span class="sxs-lookup"><span data-stu-id="71613-121">Property</span></span>|<span data-ttu-id="71613-122">Valore</span><span class="sxs-lookup"><span data-stu-id="71613-122">Value</span></span>|  
|--------------|-----------|  
|<xref:System.Exception.InnerException%2A>|<span data-ttu-id="71613-123">`null`.</span><span class="sxs-lookup"><span data-stu-id="71613-123">`null`.</span></span>|  
|<xref:System.Exception.Message%2A>|<span data-ttu-id="71613-124">La stringa di messaggio di errore fornito dal sistema.</span><span class="sxs-lookup"><span data-stu-id="71613-124">The system-supplied error message string.</span></span>|  
  
 ]]></format>
        </remarks>
        <altmember cref="T:System.Threading.Semaphore" />
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public SemaphoreFullException (string message);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(string message) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.Threading.SemaphoreFullException.#ctor(System.String)" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (message As String)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; SemaphoreFullException(System::String ^ message);" />
      <MemberSignature Language="F#" Value="new System.Threading.SemaphoreFullException : string -&gt; System.Threading.SemaphoreFullException" Usage="new System.Threading.SemaphoreFullException message" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Threading</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.10.0</AssemblyVersion>
        <AssemblyVersion>4.1.0.0</AssemblyVersion>
        <AssemblyVersion>4.1.1.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>System</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="message" Type="System.String" />
      </Parameters>
      <Docs>
        <param name="message">
          <span data-ttu-id="71613-125">Messaggio di errore in cui viene indicato il motivo dell'eccezione.</span>
          <span class="sxs-lookup">
            <span data-stu-id="71613-125">The error message that explains the reason for the exception.</span>
          </span>
        </param>
        <summary>
          <span data-ttu-id="71613-126">Inizializza una nuova istanza della classe <see cref="T:System.Threading.SemaphoreFullException" /> con un messaggio di errore specificato.</span>
          <span class="sxs-lookup">
            <span data-stu-id="71613-126">Initializes a new instance of the <see cref="T:System.Threading.SemaphoreFullException" /> class with a specified error message.</span>
          </span>
        </summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="71613-127">Il contenuto di `message` una stringa di testo deve informare l'utente sull'eccezione.</span><span class="sxs-lookup"><span data-stu-id="71613-127">The content of `message` is a text string intended to inform the user about the exception.</span></span> <span data-ttu-id="71613-128">Il chiamante del costruttore deve garantire che la stringa sia stata localizzata in base alle impostazioni cultura correnti del sistema.</span><span class="sxs-lookup"><span data-stu-id="71613-128">The caller of this constructor is required to ensure that this string has been localized for the current system culture.</span></span>  
  
 <span data-ttu-id="71613-129">Nella tabella seguente mostra i valori iniziali delle proprietà di un'istanza di <xref:System.Threading.SemaphoreFullException> inizializzata con questo costruttore di classe.</span><span class="sxs-lookup"><span data-stu-id="71613-129">The following table shows the initial property values for an instance of the <xref:System.Threading.SemaphoreFullException> class initialized with this constructor.</span></span>  
  
|<span data-ttu-id="71613-130">Proprietà</span><span class="sxs-lookup"><span data-stu-id="71613-130">Property</span></span>|<span data-ttu-id="71613-131">Valore</span><span class="sxs-lookup"><span data-stu-id="71613-131">Value</span></span>|  
|--------------|-----------|  
|<xref:System.Exception.InnerException%2A>|<span data-ttu-id="71613-132">`null`.</span><span class="sxs-lookup"><span data-stu-id="71613-132">`null`.</span></span>|  
|<xref:System.Exception.Message%2A>|<span data-ttu-id="71613-133">`message`.</span><span class="sxs-lookup"><span data-stu-id="71613-133">`message`.</span></span>|  
  
 ]]></format>
        </remarks>
        <altmember cref="T:System.Threading.Semaphore" />
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="protected SemaphoreFullException (System.Runtime.Serialization.SerializationInfo info, System.Runtime.Serialization.StreamingContext context);" />
      <MemberSignature Language="ILAsm" Value=".method familyhidebysig specialname rtspecialname instance void .ctor(class System.Runtime.Serialization.SerializationInfo info, valuetype System.Runtime.Serialization.StreamingContext context) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.Threading.SemaphoreFullException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)" />
      <MemberSignature Language="VB.NET" Value="Protected Sub New (info As SerializationInfo, context As StreamingContext)" />
      <MemberSignature Language="C++ CLI" Value="protected:&#xA; SemaphoreFullException(System::Runtime::Serialization::SerializationInfo ^ info, System::Runtime::Serialization::StreamingContext context);" />
      <MemberSignature Language="F#" Value="new System.Threading.SemaphoreFullException : System.Runtime.Serialization.SerializationInfo * System.Runtime.Serialization.StreamingContext -&gt; System.Threading.SemaphoreFullException" Usage="new System.Threading.SemaphoreFullException (info, context)" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>System.Threading</AssemblyName>
        <AssemblyVersion>4.1.0.0</AssemblyVersion>
        <AssemblyVersion>4.1.1.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>System</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="info" Type="System.Runtime.Serialization.SerializationInfo" />
        <Parameter Name="context" Type="System.Runtime.Serialization.StreamingContext" />
      </Parameters>
      <Docs>
        <param name="info">
          <span data-ttu-id="71613-134">Oggetto <see cref="T:System.Runtime.Serialization.SerializationInfo" /> che contiene i dati di oggetto serializzati relativi all'eccezione generata.</span>
          <span class="sxs-lookup">
            <span data-stu-id="71613-134">The <see cref="T:System.Runtime.Serialization.SerializationInfo" /> object that holds the serialized object data about the exception being thrown.</span>
          </span>
        </param>
        <param name="context">
          <span data-ttu-id="71613-135">Oggetto <see cref="T:System.Runtime.Serialization.StreamingContext" /> contenente le informazioni contestuali sull'origine o sulla destinazione.</span>
          <span class="sxs-lookup">
            <span data-stu-id="71613-135">The <see cref="T:System.Runtime.Serialization.StreamingContext" /> object that contains contextual information about the source or destination.</span>
          </span>
        </param>
        <summary>
          <span data-ttu-id="71613-136">Inizializza una nuova istanza della classe <see cref="T:System.Threading.SemaphoreFullException" /> con dati serializzati.</span>
          <span class="sxs-lookup">
            <span data-stu-id="71613-136">Initializes a new instance of the <see cref="T:System.Threading.SemaphoreFullException" /> class with serialized data.</span>
          </span>
        </summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="71613-137">Questo costruttore viene chiamato durante la deserializzazione per ricostituire l'oggetto eccezione trasmesso su un flusso.</span><span class="sxs-lookup"><span data-stu-id="71613-137">This constructor is called during deserialization to reconstitute the exception object transmitted over a stream.</span></span> <span data-ttu-id="71613-138">Per altre informazioni, vedere [serializzazione SOAP e XML](~/docs/standard/serialization/xml-and-soap-serialization.md).</span><span class="sxs-lookup"><span data-stu-id="71613-138">For more information, see [XML and SOAP Serialization](~/docs/standard/serialization/xml-and-soap-serialization.md).</span></span>  
  
 ]]></format>
        </remarks>
        <altmember cref="T:System.Threading.Semaphore" />
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public SemaphoreFullException (string message, Exception innerException);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(string message, class System.Exception innerException) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.Threading.SemaphoreFullException.#ctor(System.String,System.Exception)" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (message As String, innerException As Exception)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; SemaphoreFullException(System::String ^ message, Exception ^ innerException);" />
      <MemberSignature Language="F#" Value="new System.Threading.SemaphoreFullException : string * Exception -&gt; System.Threading.SemaphoreFullException" Usage="new System.Threading.SemaphoreFullException (message, innerException)" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Threading</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.10.0</AssemblyVersion>
        <AssemblyVersion>4.1.0.0</AssemblyVersion>
        <AssemblyVersion>4.1.1.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>System</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="message" Type="System.String" />
        <Parameter Name="innerException" Type="System.Exception" />
      </Parameters>
      <Docs>
        <param name="message">
          <span data-ttu-id="71613-139">Messaggio di errore in cui viene indicato il motivo dell'eccezione.</span>
          <span class="sxs-lookup">
            <span data-stu-id="71613-139">The error message that explains the reason for the exception.</span>
          </span>
        </param>
        <param name="innerException">
          <span data-ttu-id="71613-140">Eccezione che ha determinato l'eccezione corrente.</span>
          <span class="sxs-lookup">
            <span data-stu-id="71613-140">The exception that is the cause of the current exception.</span>
          </span>
          <span data-ttu-id="71613-141">Se il parametro <c>innerException</c> non è <see langword="null" />, l'eccezione corrente viene generata in un blocco <see langword="catch" /> in cui viene gestita l'eccezione interna.</span>
          <span class="sxs-lookup">
            <span data-stu-id="71613-141">If the <c>innerException</c> parameter is not <see langword="null" />, the current exception is raised in a <see langword="catch" /> block that handles the inner exception.</span>
          </span>
        </param>
        <summary>
          <span data-ttu-id="71613-142">Inizializza una nuova istanza della classe <see cref="T:System.Threading.SemaphoreFullException" /> con un messaggio di errore specificato e un riferimento all'eccezione interna che è la causa dell'eccezione corrente.</span>
          <span class="sxs-lookup">
            <span data-stu-id="71613-142">Initializes a new instance of the <see cref="T:System.Threading.SemaphoreFullException" /> class with a specified error message and a reference to the inner exception that is the cause of this exception.</span>
          </span>
        </summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="71613-143">Un'eccezione generata come risultato diretto di un'eccezione precedente deve includere un riferimento all'eccezione precedente nella proprietà <xref:System.Exception.InnerException%2A>.</span><span class="sxs-lookup"><span data-stu-id="71613-143">An exception that is thrown as a direct result of a previous exception should include a reference to the previous exception in the <xref:System.Exception.InnerException%2A> property.</span></span> <span data-ttu-id="71613-144">La proprietà <xref:System.Exception.InnerException%2A> restituisce lo stesso valore passato al costruttore oppure `null` nel caso in cui la proprietà <xref:System.Exception.InnerException%2A> non fornisca il valore di eccezione interna al costruttore.</span><span class="sxs-lookup"><span data-stu-id="71613-144">The <xref:System.Exception.InnerException%2A> property returns the same value that is passed into the constructor, or `null` if the <xref:System.Exception.InnerException%2A> property does not supply the inner exception value to the constructor.</span></span>  
  
 ]]></format>
        </remarks>
        <altmember cref="T:System.Threading.Semaphore" />
      </Docs>
    </Member>
  </Members>
</Type>